File: javascript/models/review.js
class Review {
  constructor(title, description, customerId = null, restaurantId = null) {
    this.id = Store.reviewCount++
    this.title = title
    this.description  = description
    this.customer_id = customerId
    this.restaurant_id = restaurantId

    //add Customer to Store upon initialization
    Store.reviews.push(this)
    console.log(`New Review - ${this.title} ${this.description}`)
  }

  customer() {
    console.log(Customer.prototype.findById(this.customer_id))
  }

  restaurant() {
    console.log(Restaurant.prototype.findById(this.restaurant_id))
  }
}

File: object_relations/app/models/review.rb
class Review
  attr_accessor :customer_id, :restaurant_id

  @@reviews = []
  @@counter = 0

  def initialize(customer_id, restaurant_id)
    @id = @@counter += 1
    @customer_id = customer_id
    @restaurant_id = restaurant_id
    self.all << self
  end

  def self.all
    @@reviews
  end

  def customer
    #return customer based off customer_id
    Customer.find_by_id(self.customer_id)
  end

  def restaurant
    Restaurant.find_by_id(self.restaurant_id)
  end
end
